build/mixxx.py

```diff
@@ -39,8 +39,8 @@ def __init__(self, target, machine, build, toolchain, available_features=[]):
         if build is None:
             build = 'debug'
 
-        if not build in ['debug', 'release']:
-            raise Exception("invalid build type")
+        if not build in ['debug', 'release', 'none']:
```

Comment:
This was a request from Nico, I was requesting a mode without any compiler flags. 
What happens and if it is useful or not, is up to the caller. 
It should emulate the cMake feature: 
CMAKE_BUILD_TYPE = None   (CMAKE_CXX_FLAGS or CMAKE_C_FLAGS used) 
I think it should be a release mode by default.